ti/savings/savings.go

```diff
@@ -0,0 +1,31 @@
+package savings
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	tiCfg "github.com/harness/lite-engine/ti/config"
+	"github.com/harness/lite-engine/ti/savings/cache"
+	"github.com/harness/ti-client/types"
+	"github.com/sirupsen/logrus"
+)
+
+func ParseAndUploadSavings(ctx context.Context, workspace string, log *logrus.Logger, stepID string, tiConfig *tiCfg.Cfg) {
+	// Cache Savings
+	start := time.Now()
+	state, timeTaken, err := cache.ParseCacheSavings(workspace, log)
```

Comment:
For TI timesavings we use the time taken by the step, but over here we are getting build time from the gradle profile report. 
What approach should be used for build cache? For build cache will the report time be same as the time taken for the step?

Reply:
Yes that's the approach we want to take. Gradle build time is what matters for its time savings.